blocks/extensions/saveload.py

```diff
@@ -85,83 +88,78 @@ def do(self, callback_name, *args):
         construction stage.
 
         """
-        from_main_loop, from_user = self.parse_args(callback_name, args)
+        _, from_user = self.parse_args(callback_name, args)
         try:
             path = self.path
             if from_user:
                 path, = from_user
-            already_saved_to = self.main_loop.log.current_row.get(SAVED_TO, ())
-            self.main_loop.log.current_row[SAVED_TO] = (
-                already_saved_to + (path,))
-            secure_pickle_dump(self.main_loop, path)
+            secure_dump(self.main_loop, path, use_cpickle=self.use_cpickle)
             filenames = self.save_separately_filenames(path)
             for attribute in self.save_separately:
-                secure_pickle_dump(getattr(self.main_loop, attribute),
-                                   filenames[attribute])
+                secure_dump(getattr(self.main_loop, attribute),
+                            filenames[attribute], cPickle.dump)
         except Exception:
-            self.main_loop.log.current_row[SAVED_TO] = None
+            path = None
             raise
+        finally:
+            already_saved_to = self.main_loop.log.current_row.get(SAVED_TO, ())
+            self.main_loop.log.current_row[SAVED_TO] = (already_saved_to +
+                                                        (path,))
 
 
-class LoadFromDump(TrainingExtension):
-    """Loads a dump into the main loop.
+class Load(TrainingExtension):
+    """Loads a saved checkpoint into the main loop.
 
     Makes a `LOADED_FROM` record in the log with the dump path.
 
     Parameters
     ----------
-    state_path : str
+    path : str
         The path to the folder with dump.
+    load_iteration_state : bool
+        If `True`, load the iteration state. This can be useful when your
+        model has very long epochs, and you want to resume when you were in
+        the middle of one. Defaults to `False`.
+    load_log : bool
+        If `True`, load the old log and continue logging from there.
+        Convenient because you end up with a single log of the entire
+        training history. Defaults to `False`.
 
     Notes
     -----
-    Requires the model to be a Brick or a list of Bricks.
+    Requires the model to be created entirely using bricks, with a unique
+    path/name for each brick, so that the parameters can be matched to
+    their values.
+
+    In order to load the iteration state and the log, the saved model needs
```

Comment:
For future work I would suggest to pickle log separately. We should be able to do it even without using  persistent id. Simply deleting the log attribute right before dumping and adding it back when loading should do the job.

But so far I am fine with merging the current variant.